### ISSUE TITLE
Compatibility with urls containing IPv6  IP addresses

### ISSUE THREAD

**Issue by answerquest** [usebruno/bruno#2338]
### I have checked the following:

- [X] I've searched existing issues and found nothing related to my issue.

### Describe the feature you want to add

I have a server on an IPv6 address, and it has deployed an api on a port. There is a GET api endpoint active on it. The following api url works fine from browser as well as Postman:  
```
http://[some:ip:v6:ip:address]:8000/api
```
But the same url in Bruno when hit as a GET api, returns this error:
```
Error invoking remote method 'send-http-request': Error: getaddrinfo ENOTFOUND [some:ip:v6:ip:address]
```

Desired outcome: the api on the IPv6 address should work properly in Bruno same way it works from web browser or Postman.

Note: My actual IPv6 address is not revealed as it's a work thing and I'm not at liberty to publicly expose it.

PS: enclosing the IPv6 IP address within square brackets `[]` is the standard practice right now. I'm even able to connect to this server from FileZilla (SFTP) by giving the host enclosed in `[]`

### Mockups or Images of the feature

![Screenshot from 2024-05-18 21-10-22](https://github.com/usebruno/bruno/assets/9089552/43b61527-3419-4226-9412-ec8e0a6d5612)  
screenshot of the IPv6 api call working successfully in Postman

**Comment by gemiusz** [comment]
Bruno: 1.18.1
System: Windows 11

TL;DR; Can't reproduce this bug.

@answerquest I have try to reproduce this bug but it's failed and in my opinion it's something with resolving DNS/IP on your side.
You must provide reproducable steps to confirm this bug.

Steps what I use to try to reproduce:
- preprare Wiremock and run it:
  - https://wiremock.org/docs/download-and-installation/  -> https://repo1.maven.org/maven2/org/wiremock/wiremock-standalone/3.6.0/wiremock-standalone-3.6.0.jar
  - prepared mapper:
```
{
    "request": {
        "method": "GET",
        "url": "/4test"
    },
    "response": {
		"headers": {
            "Content-Type": "application/json"
        },
        "status": 200,
        "body": "{\"status\":\"GOOD\",\"fruits\":[{\"name\":\"Apple\",\"details\":{\"size\":500}},{\"name\":\"Cherry\",\"details\":{\"size\":100}}]}"
    }
}
```
  - run: ```java -jar .\wiremock-standalone-3.6.0.jar --verbose```


- make call by localhost  -> [::1]:
  - Bruno:
![image](https://github.com/usebruno/bruno/assets/30750006/03d20db9-36a5-4de8-a69b-2719117914c3)

  - Wiremock:
```
2024-06-06 21:51:46.097 Request received:
[0:0:0:0:0:0:0:1] - GET /4test

Accept: [application/json, text/plain, */*]
request-start-time: [1717703506094]
User-Agent: [axios/1.7.2]
Accept-Encoding: [gzip, compress, deflate, br]
Host: [[::1]:8080]
Connection: [close]



Matched response definition:
{
  "status" : 200,
  "body" : "{\"status\":\"GOOD\",\"fruits\":[{\"name\":\"Apple\",\"details\":{\"size\":500}},{\"name\":\"Cherry\",\"details\":{\"size\":100}}]}",
  "headers" : {
    "Content-Type" : "application/json"
  }
}

Response:
HTTP/1.1 200
Content-Type: [application/json]
Matched-Stub-Id: [dc057c5e-ab26-4e64-9c45-48b1e2728924]

```



- make call by Link-local IPv6 Address - interface 1:
![image](https://github.com/usebruno/bruno/assets/30750006/ce3602fc-7989-41d2-81a5-f9c0c6f50186)
![image](https://github.com/usebruno/bruno/assets/30750006/90ed23ba-09ce-45fb-abc4-30d5c918a449)

- make call by Link-local IPv6 Address - interface 2:
![image](https://github.com/usebruno/bruno/assets/30750006/3f9a1a34-df3a-456d-a980-79fc336f1caf)
![image](https://github.com/usebruno/bruno/assets/30750006/18b34ea6-5890-460d-b24a-adbb991118c8)

- make call by Link-local IPv6 Address - interface 3:
![image](https://github.com/usebruno/bruno/assets/30750006/5dd532f0-7e17-4b1e-8d43-8023c88c3d83)
![image](https://github.com/usebruno/bruno/assets/30750006/77c67e85-3d6f-483b-a988-2efb3d163994)

**Comment by kinuax** [comment]
I can reproduce the same `Error invoking remote method 'send-http-request': Error: getaddrinfo ENOTFOUND [ipv6]` with Bruno 1.17.0 and 1.18.0 on Linux. Besides, corresponding browser, `curl` and `httpx` requests work correctly.

**Comment by answerquest** [comment]
@gemiusz  Hi I should have shared my OS details, maybe this is happening by the way Bruno interacts with the OS? 

OS where bug is happening: Linux Mint 21.1

VERSION_CODENAME=vera
UBUNTU_CODENAME=jammy

CPU architecture: x86_64 / amd64

Everything else on the system is able to hit IPv6 addresses just fine, including Postman for linux.

Also my system's own IP address as assigned by internet service provider (not a fixed one) is an IPv6 one : `2401:4900:1c17:a73b:aae4:xxxx:xxxx:xxxx` (masking last parts)

**Comment by tomaszkoziara** [comment]
I'm having the same problem on macOS 14.6.1. curl works, Firefox works, but bruno doesn't.

**Comment by mzammit-ot** [comment]
This issue is blocking the verification of IPv6 endpoints in our microservices test environment. The behaviour appears to mimic an issue recently fixed in Axios:

[https://github.com/axios/axios/pull/5731](https://github.com/axios/axios/pull/5731)

The IPv6 Literal used is valid, and can can be correctly resolved via the MS DNS instance configured for the isolated vNet. 

**Bruno System Details**
| Field | Value |
| --- | --- |
| OS Name | `Microsoft Windows Server 2022 Standard` |
| Version | `10.0.20348 Build 20348` |
| Bruno IP Address | `[FC01:0002:0003:0004:0250:56FF:FE0A:D71F]` |
| Target URL | `http://[FC01:0002:0003:0004:0250:56FF:FE0A:D721]:3356` |
| Bruno | `1.32.0` | 
| Electron | `31.2.1` |
| Chrome | `126.0.6478.127` |
| node | `20.15.0` |
| v8 | `12.6.228.21-electron.0` |

**Comment by That-1-QA** [comment]
> This issue is blocking the verification of IPv6 endpoints in our microservices test environment. The behaviour appears to mimic an issue recently fixed in Axios:
> 
> [axios/axios#5731](https://github.com/axios/axios/pull/5731)
> 
> The IPv6 Literal used is valid, and can can be correctly resolved via the MS DNS instance configured for the isolated vNet.
> 
> **Bruno System Details**
> Field 	Value
> OS Name 	`Microsoft Windows Server 2022 Standard`
> Version 	`10.0.20348 Build 20348`
> Bruno IP Address 	`[FC01:0002:0003:0004:0250:56FF:FE0A:D71F]`
> Target URL 	`http://[FC01:0002:0003:0004:0250:56FF:FE0A:D721]:3356`
> Bruno 	`1.32.0`
> Electron 	`31.2.1`
> Chrome 	`126.0.6478.127`
> node 	`20.15.0`
> v8 	`12.6.228.21-electron.0`

Hmm, that Axios issue may be the cause. Looking at the package.json for the bruno app shows a 1.7.5 version:
https://github.com/usebruno/bruno/blob/32153c4dbf5652f4a48ef8f27f21d8206e6b1655/packages/bruno-app/package.json#L23C4-L23C22

It looks like that Axios issue with handling of IPv6 literals was resolved in 1.7.7.  If Axios is being used by the Bruno App to make requests, then this is likely the root cause.

For what it's worth, the bruno-cli would likely be affected too:
https://github.com/usebruno/bruno/blob/32153c4dbf5652f4a48ef8f27f21d8206e6b1655/packages/bruno-cli/package.json#L32

**Comment by helloanoop** [comment]
Just merged the PR - #3878 
The fix should be available in today's nightly build to test.
A release is scheduled to go out next week with these fixes.

**Comment by sreelakshmi-bruno** [comment]
Hi, here is the [nightly build](https://github.com/usebruno/bruno-nightly-builds/tree/main) to test the fix. Please check it out and let me know how it goes. Thanks!

**Comment by mzammit-ot** [comment]
> Hi, here is the [nightly build](https://github.com/usebruno/bruno-nightly-builds/tree/main) to test the fix. Please check it out and let me know how it goes. Thanks!

Hi @sreelakshmi-bruno 
I downloaded, installed, and ran the latest nightly build:
| Bruno | 1.39.0 | 
| --- | --- |
| electron | 33.2.1 | 
| chrome | 130.0.6723.137 | 
| node | 20.18.1 | 
| v8 | 13.0.245.19-electron.0 | 

However, the behaviour remains the same, with an error being returned: `Error invoking remote method 'send-http-request': Error: connect ECONNREFUSED fc01:2:3:4:250:56ff:fe0a:d721:3356`

For context, I am performing a 'GET' request to `{{publication-service-origin}}/publication/api/v1/health` where the collection variable `{{publication-service-origin}}` is set to `http://[FC01:2:3:4:0250:56FF:FE0A:D721]:3356`

The VM running Bruno is:
| Edition | Windows Server 2022 Standard | 
| --- | --- |
| Version | 21H2 | 
| Installed on | 9/‎22/‎2021 | 
| OS build | 20348.1547 | 

Please let me know if there are additional files required or steps I can perform in order to provide additional information.

MZ

**Comment by naman-bruno** [comment]
Hi @mzammit-ot,
Thank you for your detailed feedback!
We tried to reproduce the issue on our end but were unable to encounter the same behavior. Could you also try making the request using curl and check if the IPv6 address is resolving correctly? This will help us determine if the issue is specific to Bruno or if it might be network-related.

Let us know what you find, and we’ll be happy to investigate further!
